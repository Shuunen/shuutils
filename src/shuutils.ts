export * from './analytics'
export * from './array-insert-after'
export * from './array-pick-one'
export * from './array-remove-value'
export * from './array-shuffle'
export * from './array-unique'
export * from './browser-clipboard'
export * from './browser-fill-like-human'
export * from './browser-inject-styles'
export * from './browser-page-change'
export * from './browser-routing'
export * from './browser-scout'
export * from './browser-styles'
export * from './browser-toast'
export * from './colors'
export * from './constants'
export * from './date-relative'
export * from './dates'
export * from './debounce'
export * from './dom'
export * from './environment'
export * from './events'
export * from './expect-type'
export * from './flags'
export * from './functions'
export * from './logger'
export * from './memoize'
export * from './number-ranged-score'
export * from './number-round'
export * from './object-cap-nesting'
export * from './object-clone'
export * from './object-equal'
export * from './object-flatten'
export * from './object-serializer'
export * from './object-sort'
export * from './objects'
export * from './random'
export * from './result'
export * from './state'
export * from './storage'
export * from './string-case'
export * from './string-remove-accents'
export * from './strings'
export * from './system'
export * from './testing-align-for-snap'
export * from './throttle'
export * from './timestamps'
export * from './translate'
export type * from './types'
