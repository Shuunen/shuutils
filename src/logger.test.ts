import { expect, it } from 'vitest'
import { Logger, red } from './shuutils'

it('logger A', () => {
  const loggerA = new Logger()
  expect(loggerA.options.isActive).toBe(true)
  loggerA.info('This info 0 should be logged')
})

it('logger B', () => {
  const loggerB = new Logger({ isActive: false, willLogDelay: false, willOutputToConsole: false, willOutputToMemory: true })
  expect(loggerB.options.isActive).toBe(false)
  loggerB.info('This info 1 should not be logged')
  loggerB.options.isActive = true
  loggerB.info('This info 2 should be logged')
  loggerB.info('This info 3 should be logged too')
  loggerB.success('This success 0 should be logged')
  loggerB.warn('This warn 1 should be logged')
  loggerB.fix('This fix 1 should be logged', 42)
  loggerB.error('This error 0 should be logged', { isKeyA: true, keyB: 'John' })
  loggerB.error(new Error('This error 1 should be logged too'))
  expect(loggerB.inMemoryLogs, 'loggerB inMemoryLogs').toMatchSnapshot()
})

it('logger C', () => {
  const loggerC = new Logger({ minimumLevel: '7-error', willLogDate: true, willLogDelay: true, willLogTime: true, willOutputToConsole: false, willOutputToMemory: true })
  loggerC.warn('This warn 2 should not be logged')
  loggerC.success('This success 1 should not be logged')
  loggerC.error('This error 1 should be logged')
  loggerC.disable()
  loggerC.error('This error 2 should not be logged')
  loggerC.enable()
  loggerC.error('This error 3 should be logged')
  loggerC.test(true, 'This test 1 should not be logged')
  loggerC.options.minimumLevel = '2-test'
  loggerC.test(true, 'This test 2 should be logged')
  loggerC.test(false, 'This test 3 should be logged')
  // cannot use checkSnapshot because of the date
  expect(loggerC.inMemoryLogs.length, 'loggerC has 4 inMemoryLogs').toBe(4)
})

it('logger D', () => {
  const loggerD = new Logger({ willLogDelay: false, willOutputToConsole: false, willOutputToMemory: true })
  loggerD.info('This info 4 should be logged', 12)
  loggerD.info('This info 5 should be logged too', [1, 2, 3])
  loggerD.warn('This warn 3 should be logged', { isKeyC: true, keyA: 1, keyB: 'John' })
  loggerD.disable()
  loggerD.warn('This warn 4 should not be logged')
  loggerD.error('This error 4 should not be logged')
  loggerD.enable()
  loggerD.error('This error 5 should be logged', null) // eslint-disable-line unicorn/no-null
  loggerD.test(true, 'This test 4 should be logged', undefined) // eslint-disable-line unicorn/no-useless-undefined
  loggerD.test(false, 'This test 5 should be logged', () => 'Hello world')
  loggerD.debug('This debug 1 should be logged', true, [], {})
  loggerD.options.minimumLevel = '3-info'
  loggerD.debug('This debug 2 should not be logged')
  expect(loggerD.inMemoryLogs, 'loggerD inMemoryLogs').toMatchSnapshot()
})

const loggerE = new Logger({ willOutputToConsole: false })
it('logger clean A', () => {
  expect(loggerE.clean()).toBe('')
})
it('logger clean B', () => {
  expect(loggerE.clean(red("Oh I'm in red now ?!"))).toMatchInlineSnapshot('"Oh I\'m in red now ?!"')
})
it('logger clean C', () => {
  expect(loggerE.clean('an array ?', [12, 42])).toMatchInlineSnapshot('"an array ? [12,42]"')
})
it('logger clean D', () => {
  expect(loggerE.clean('a function ?', () => 'Hello world')).toMatchInlineSnapshot('"a function ? () => \'Hello world\'"')
})
it('logger clean E', () => {
  expect(loggerE.clean('an object ?', { isFull: true, keyA: 1, keyB: 'John' })).toMatchInlineSnapshot("\"an object ? {'isFull':true,'keyA':1,'keyB':'John'}\"")
})
it('logger clean F', () => {
  expect(loggerE.clean('a boolean ?', true)).toMatchInlineSnapshot('"a boolean ? true"')
})
it('logger clean G', () => {
  expect(loggerE.clean('a number ?', 42)).toMatchInlineSnapshot('"a number ? 42"')
})
it('logger clean H', () => {
  expect(loggerE.clean('a string ?', 'Hello world')).toMatchInlineSnapshot('"a string ? Hello world"')
})
it('logger clean I', () => {
  // eslint-disable-next-line unicorn/no-null
  expect(loggerE.clean('a null ?', null)).toMatchInlineSnapshot('"a null ? null"')
})
it('logger clean J', () => {
  // eslint-disable-next-line unicorn/no-useless-undefined
  expect(loggerE.clean('an undefined ?', undefined)).toMatchInlineSnapshot('"an undefined ? undefined"')
})
it('logger clean K', () => {
  expect(loggerE.clean('a date ?', new Date('2020-01-01'))).toMatchInlineSnapshot('"a date ? \'2020-01-01T00:00:00.000Z\'"')
})
it('logger clean L', () => {
  expect(loggerE.clean('a regexp ?', /Hello world{3,5}/u)).toMatchInlineSnapshot('"a regexp ? {}"')
}) // not supported for now

it('logger F show', () => {
  const loggerF = new Logger({ willLogDelay: false, willOutputToConsole: false, willOutputToMemory: true })
  loggerF.showInfo('This info 1 should be logged', 12)
  loggerF.info('This info 2 should be logged too', [1, 2, 3])
  loggerF.warn('This warn 3 should be logged', { isKeyC: true, keyA: 1, keyB: 'John' })
  loggerF.showError('This error 4 should not be logged')
  loggerF.error('This error 5 should be logged', null) // eslint-disable-line unicorn/no-null
  loggerF.showSuccess('This success 6 should be logged', 12)
  loggerF.test(true, 'This test 7 should be logged', undefined) // eslint-disable-line unicorn/no-useless-undefined
  loggerF.test(false, 'This test 8 should be logged', () => 'Hello world')
  loggerF.debug('This debug 9 should be logged', true, [], {})
  loggerF.options.minimumLevel = '3-info'
  loggerF.debug('This debug 10 should not be logged')
  expect(loggerF.inMemoryLogs, 'loggerF inMemoryLogs').toMatchSnapshot()
})
