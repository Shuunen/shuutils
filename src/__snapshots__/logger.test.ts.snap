// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`logger B 1`] = `
[
  " info This info 2 should be logged",
  " info This info 3 should be logged too",
  " good This success 0 should be logged",
  " warn This warn 1 should be logged",
  "  fix This fix 1 should be logged 42",
  "error This error 0 should be logged {'isKeyA':true,'keyB':'John'}",
  "error This error 1 should be logged too",
]
`;

exports[`logger D 1`] = `
[
  " info This info 4 should be logged 12",
  " info This info 5 should be logged too [1,2,3]",
  " warn This warn 3 should be logged {'isKeyC':true,'keyA':1,'keyB':'John'}",
  "error This error 5 should be logged null",
]
`;

exports[`logger F show 1`] = `
[
  " info This info 1 should be logged 12",
  " info This info 2 should be logged too [1,2,3]",
  " warn This warn 3 should be logged {'isKeyC':true,'keyA':1,'keyB':'John'}",
  "error This error 4 should not be logged",
  "error This error 5 should be logged null",
  " good This success 6 should be logged 12",
]
`;
